src/lib/openapi/spec/client-metrics-schema.test.ts

```diff
@@ -0,0 +1,44 @@
+import { validateSchema } from '../validate';
+import { ClientMetricsSchema } from './client-metrics-schema';
+
+test('clientMetricsSchema full', () => {
+    const data: ClientMetricsSchema = {
+        appName: 'a',
+        instanceId: 'some-id',
+        environment: 'some-env',
+        bucket: {
+            start: Date.now(),
+            stop: Date.now(),
+            toggles: {
+                someToggle: {
+                    yes: 52,
+                    no: 2,
+                    variants: {},
+                },
+            },
+        },
+    };
+
+    expect(
+        validateSchema('#/components/schemas/clientMetricsSchema', data),
+    ).toBeUndefined();
+});
+
+test('clientMetricsSchema unexpected input', () => {
```

Comment:
We should add a few e2e tests as well with various unexpected/partial/missing metrics input. 

Should also figure out what we should respond with in those cases (what constitutes a crash). 

Should we respond with 400 when the schema fails, or stick to 20x responses even then?

Reply:
I think I would be expecting a 400 if the data I am sending is wrong